### PR TITLE
Add new OpenAI embedding models + dimensions parameter

### PR DESCRIPTION
**Pull request by Mitch528**
This PR adds the new OpenAI embedding models, as announced here:

https://openai.com/blog/new-embedding-models-and-api-updates

Also added the new `dimensions` parameter described in the announcement.

- https://github.com/openai/openai-openapi/pull/179

### CODE REVIEW

File: OpenAI-DotNet/Models/Model.cs [hunk 1]
```diff
@@ -130,6 +130,10 @@ public Model(string id, string ownedBy = null)
         /// The default model for <see cref="Embeddings.EmbeddingsEndpoint"/>.
         /// </summary>
         public static Model Embedding_Ada_002 { get; } = new("text-embedding-ada-002", "openai");
+        
+        public static Model Embedding_3_Small { get; } = new("text-embedding-3-small", "openai");
+        
+        public static Model Embedding_3_Large { get; } = new("text-embedding-3-large", "openai");
```
Comment (StephenHodgson) [inline OpenAI-DotNet/Models/Model.cs:136]: lets add descriptions for these models.

File: OpenAI-DotNet/Embeddings/EmbeddingsEndpoint.cs [hunk 2]
```diff
@@ -54,11 +58,15 @@ public async Task<EmbeddingsResponse> CreateEmbeddingAsync(string input, string
...
+        /// Only supported in text-embedding-3 and later models
+        /// </param>
         /// <param name="cancellationToken">Optional, <see cref="CancellationToken"/>.</param>
         /// <returns><see cref="EmbeddingsResponse"/></returns>
-        public async Task<EmbeddingsResponse> CreateEmbeddingAsync(IEnumerable<string> input, string model = null, string user = null, CancellationToken cancellationToken = default)
-            => await CreateEmbeddingAsync(new EmbeddingsRequest(input, model, user), cancellationToken).ConfigureAwait(false);
-
+        public async Task<EmbeddingsResponse> CreateEmbeddingAsync(IEnumerable<string> input, string model = null, string user = null, int? dimensions = null, CancellationToken cancellationToken = default)
+            => await CreateEmbeddingAsync(new EmbeddingsRequest(input, model, user, dimensions), cancellationToken).ConfigureAwait(false);
+
```
Comment (StephenHodgson) [inline OpenAI-DotNet/Embeddings/EmbeddingsEndpoint.cs:69]:
```suggestion

```

File: OpenAI-DotNet-Tests/TestFixture_06_Embeddings.cs [hunk 3]
```diff
@@ -13,6 +14,17 @@ public async Task Test_1_CreateEmbedding()
             Assert.IsNotNull(embedding);
             Assert.IsNotEmpty(embedding.Data);
         }
+        
+        [Test]
+        public async Task Test_1_CreateEmbeddingWithDimensions()
```
Comment (StephenHodgson) [inline OpenAI-DotNet-Tests/TestFixture_06_Embeddings.cs:19]:
```suggestion
        public async Task Test_2_CreateEmbeddingWithDimensions()
```

**Review by StephenHodgson — CHANGES_REQUESTED** [review]
See comments

**Comment by Mitch528** [comment]
> See comments

updated!

**Review by StephenHodgson — APPROVED** [review]

**Pull request merged**